letsencrypt/storage.py

```diff
@@ -252,6 +252,7 @@ def __init__(self, config_filename, cli_config):
         self.privkey = self.configuration["privkey"]
         self.chain = self.configuration["chain"]
         self.fullchain = self.configuration["fullchain"]
+        self.live_dir = os.path.dirname(self.cert)
```

Comment:
I don't think you should change the structure of the `live_dir` layout; my sense is that the things djb is asking for would all involve reorganising `archive/` but leaving the links in `live/` as they currently are (though maybe more kinds of file should be added there...)